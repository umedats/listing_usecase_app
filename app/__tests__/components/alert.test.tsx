import AlertComponent from "@/app/components/alert";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Alert component", () => {
  const stateHandler = jest.fn();

  test(" Component renders visible success", () => {
    const state = true;

    const { asFragment, getByTestId } = render(
      <AlertComponent
        id={"test"}
        state={state}
        stateHandler={stateHandler}
        severity={"success"}
        message={"success description"}
      />,
    );

    expect(getByTestId("test")).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  test(" Component renders visible error", () => {
    const openHandler = true;

    const { asFragment, getByTestId } = render(
      <AlertComponent
        id={"test"}
        state={openHandler}
        stateHandler={stateHandler}
        severity={"error"}
        message={"success description"}
      />,
    );

    expect(getByTestId("test")).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
  test(" Component renders visible info", () => {
    const openHandler = true;

    const { asFragment, getByTestId } = render(
      <AlertComponent
        id={"test"}
        state={openHandler}
        stateHandler={stateHandler}
        severity={"info"}
        message={"success description"}
      />,
    );

    expect(getByTestId("test")).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
  test(" Component renders visible warning", () => {
    const openHandler = true;

    const { asFragment, getByTestId } = render(
      <AlertComponent
        id={"test"}
        state={openHandler}
        stateHandler={stateHandler}
        severity={"warning"}
        message={"success description"}
      />,
    );

    expect(getByTestId("test")).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Component renders invisible", () => {
    const openHandler = false;

    const { asFragment, getByTestId } = render(
      <AlertComponent
        id={"test"}
        state={openHandler}
        stateHandler={stateHandler}
        severity={"success"}
        message={"success description"}
      />,
    );

    expect(getByTestId("test")).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
});
