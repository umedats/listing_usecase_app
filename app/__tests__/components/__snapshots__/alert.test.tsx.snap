// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component  Component renders visible error 1`] = `
<DocumentFragment>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-18djxyj-MuiPaper-root-MuiAlert-root"
          data-testid="test"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="ErrorOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            success description
          </div>
          <div
            class="MuiAlert-action css-ki1hdl-MuiAlert-action"
          >
            <button
              aria-label="close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert component  Component renders visible info 1`] = `
<DocumentFragment>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-ethv1g-MuiPaper-root-MuiAlert-root"
          data-testid="test"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="InfoOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            success description
          </div>
          <div
            class="MuiAlert-action css-ki1hdl-MuiAlert-action"
          >
            <button
              aria-label="close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert component  Component renders visible success 1`] = `
<DocumentFragment>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-4bzs8s-MuiPaper-root-MuiAlert-root"
          data-testid="test"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="SuccessOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            success description
          </div>
          <div
            class="MuiAlert-action css-ki1hdl-MuiAlert-action"
          >
            <button
              aria-label="close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert component  Component renders visible warning 1`] = `
<DocumentFragment>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-145tbyc-MuiPaper-root-MuiAlert-root"
          data-testid="test"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="ReportProblemOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            success description
          </div>
          <div
            class="MuiAlert-action css-ki1hdl-MuiAlert-action"
          >
            <button
              aria-label="close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert component Component renders invisible 1`] = `
<DocumentFragment>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-4bzs8s-MuiPaper-root-MuiAlert-root"
          data-testid="test"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="SuccessOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            success description
          </div>
          <div
            class="MuiAlert-action css-ki1hdl-MuiAlert-action"
          >
            <button
              aria-label="close"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
